### ISSUE TITLE
Properties of Components not shown in WebACS/ No reaction of Property Editor.

### ISSUE THREAD

**Issue by sabicalija** [asterics/WebACS#30]
# Description
Just tried the [`asterics-camerainput-cameramouse`](https://github.com/asterics/asterics-camerainput-cameramouse) and [`asterics-microcontroller-arduino`](https://github.com/asterics/asterics-microcontroller-arduino) demo.

When I open use `Edit Model -> `[Open in WebACS](http://asterics.github.io/AsTeRICS/webapps/WebACS/?areBaseURI=http://localhost:8081&openFile=http://asterics.github.io/AsTeRICS/webapps/asterics-camerainput-cameramouse/models/XFaceTrackerMouse(WLM).acs) the model is opened successfully.
Even connection and interaction with (locally) running ARE works fine.

But when I select one of the components in the model in the WebACS, there is no reaction of the property editor. I've tried it several times, both with FireFox and Chrome. And I'm able to reproduce this behavior.

# How to reproduce
1. Start the any demo by opening index.html of the respective demo repository
2. Click on [Open in WebACS](http://asterics.github.io/AsTeRICS/webapps/WebACS/?areBaseURI=http://localhost:8081&openFile=http://asterics.github.io/AsTeRICS/webapps/asterics-camerainput-cameramouse/models/XFaceTrackerMouse(WLM).acs)
3. Select any component
4. No reaction of the property editor. Configuration is not possible.

![image](https://user-images.githubusercontent.com/13609768/38058574-ac41e4f0-32e3-11e8-8587-c11adda9c886.png)

I've tried those examples on a `Ubuntu 16.4.3` inside a `virtualbox (v: 5.2.8 r121009 (Qt5.6.1))`.

# Workaround
The only way I was able to force the property editor to react was when performing following steps:
1. Select a component of the model.
2. Click on Download Model

![image](https://user-images.githubusercontent.com/13609768/38058659-ec67c96e-32e3-11e8-88d1-374e1f149b7c.png)

A new tab is opened with the model, currently running in the ARE.
When switching back to the previous tab, the properties of the component selected (at step 1) are now displayed in the property editor.

I'll test different setup/scenarios in future an report it here, so that problem can be localized and removed in future.

**Comment by sabicalija** [comment]
I've found the problem that is causing the erroneous behavior:

model.js
```js
// event 'modelChangedEvent' fired before handler registered.
537    returnObj.events.fireEvent('modelChangedEvent');
```

webACS.js
```js
// circular dependency: ACS.view (and its event handlers) should
// be called (registered) before the modelList is loaded.
// However, ACS.view expects the modelList as argument.
103    // setup the WebACS
104    modelList = ACS.modelList();
105    clipBoard = ACS.clipBoard();
106    view = ACS.view(modelList, clipBoard);	
107    ACS.areStatus.setModelList(modelList);
```

## Note

The comparison of *component selection* in a downloaded model (cf. 1st picture) and a model loaded using a query string (cf. 2nd picture) revealed that usually 2 events are fired (`deSelectEventHandler`, `deSelectedComponentEventHandler`) when a component is selected. However, only the first is fired when the model is loaded using a query string (`openFile`)

![image](https://user-images.githubusercontent.com/13609768/53785994-7ec43c00-3f1a-11e9-8324-9c6cfe2d5d68.png)

![image](https://user-images.githubusercontent.com/13609768/53785998-84218680-3f1a-11e9-8eda-cc02e67ca9aa.png)


## Call stack of `ACS.modelList()`

![image](https://user-images.githubusercontent.com/13609768/53785833-21c88600-3f1a-11e9-8213-be609be73eb7.png)

**Comment by deinhofer** [comment]
Great, you are my man!! :-)

**Comment by sabicalija** [comment]
@kingcole do you have a suggestion on how to change the loading of models provided with using a query string?

**Comment by kingcole** [comment]
OK, I see the problem - we need to wait with loading the model until the view has been constructed. What about the following: take the code loading the querystring-model (currently in the constructor of modelList.js, line 105...) and move it to a public function. Then, in the constructor of webACS.js, AFTER all the setup functions have been called (e.g. line 108), add something like 

if (ACS.openFile) modelList.loadQueryStringModel();

That way the model is loaded after the view has been constructed and hopefully all should be fine.
(could have done this myself, but I currently don't have the time to thoroughly test it and I don't want to break anything)

**Comment by sabicalija** [comment]
Fixed by https://github.com/asterics/WebACS/pull/34

**Comment by klues** [comment]
#34 is actually not merged yet, so closing this issue probably wasn't intended?!

**Comment by sabicalija** [comment]
It was intended. :slightly_smiling_face: But you're right. It should be opened until it is merged. Thanks!

**Comment by deinhofer** [comment]
fixed by #34
